### PR TITLE
[TVMScript] support kTarget func attr in tir script

### PR DESCRIPTION
**Pull request by wrongtest**
Hi, this PR propose to support `kTarget` func attr in script. Since relay backend and SplitHostDevice will append the target information to PrimFunc, it is really helpful for us (de)serializing these backend PrimFuncs.

Also glad to know whether it is the proper way to extend the grammar with `Target` object, many thanks.

### CODE REVIEW

**Comment by junrushao1994** [comment]
Thanks for the contribution! Given that target string is simply a syntactic sugar to target’s JSON-like configuration, I would suggest that we print Target to json rather than string with TargetNode::Export

**Comment by wrongtest** [comment]
@junrushao1994 Thanks for advice! Update the printer to use TargetNode::Export(). Also is the `T.target()` acceptable? Or else maybe we could just parse the dict value to Target only if key == kTarget in func attrs parsing.

File: python/tvm/script/tir/__init__.pyi [hunk 1]
```diff
@@ -371,3 +372,5 @@ class handle:
     def data(self: handle) -> Ptr: ...
 
 class Ptr: ...
+
+def target(target_str: str) -> Target: ...
```
Comment (junrushao1994) [inline python/tvm/script/tir/__init__.pyi:376]: the input type is not limited to `str`, right?

File: tests/python/unittest/test_tvmscript_roundtrip.py [hunk 2]
```diff
@@ -3205,5 +3205,30 @@ def test_nontrivial_range_axis():
     tvm.ir.assert_structural_equal(func, rt_func, True)
 
 
+@T.prim_func
+def func_with_target_spec() -> None:
+    T.func_attr(
```
Comment (junrushao1994) [inline tests/python/unittest/test_tvmscript_roundtrip.py:3210]: Shall we add a unittest for target tag? For example:

```
T.target("nvidia/nvidia-a100")
```

File: python/tvm/script/tir/special_stmt.py [hunk 3]
```diff
@@ -17,6 +17,7 @@
 """TVM Script Parser Special Stmt Classes"""
 # pylint: disable=unused-argument, no-self-argument, inconsistent-return-statements
 # pylint: disable=relative-beyond-top-level
+from ctypes import ArgumentError
```
Comment (junrushao1994) [inline python/tvm/script/tir/special_stmt.py:20]: Probably we don't need this particular kind of error from ctypes. What about we just raise ValueError?

**Review by junrushao1994 — COMMENTED** [review]
Thanks @wrongtest! Overall it's in good shape

**Comment by junrushao1994** [comment]
CC: @Hzfengsy

**Review by junrushao1994 — APPROVED** [review]
LGTM. Thank you so much for your contribution! I will leave @Hzfengsy to manage this PR, and after verifying that structural equality test does include nested function attributes, please go ahead and merge this

**Pull request merged**